props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,787 @@
+        PVLOG_NORMAL << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+        break;
+      case MODE_AUTO:
+        if (!CheckEmpty()) {
+          SelectAutoFirePair();
+          auto_firing_ = true;
+          SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+          SaberBase::DoBeginLockup();
+        }
+        break;
+    }
+  }
+
+ void DoAutoFire() override {
+    if (!SFX_auto) return;
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (CheckEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
+    // if there are different length auto.wavs, the last played's length is used.
+    if (!auto_firing_ && CheckEmpty() && trigger_is_pressed_ && (millis() - auto_time_) > auto_sound_length) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      auto_time_ = millis(); // Reset for the next empty sound
+    }
+ }
+
+  void Loop() override {
+    PropBase::Loop();
+    PollNextAction();
+    sound_library_.Poll(wav_player);
+    if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+    if (blaster_mode == MODE_AUTO) {
+      DoAutoFire();
+    }
+  }
+
+  void Reload() override {
+    Blaster::Reload();
+    is_jammed_ = false;
+  }
+
+  void ClipOut() override {
+    Blaster::ClipOut();
+    no_clip_ = true;
+  }
+
+  void ClipIn() override {
+    Blaster::ClipIn();
+    is_jammed_ = false;
+    no_clip_ = false;
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+     if (!SFX_poweron && !SaberBase::IsOn()) {
+      On();
+    }
+  }
+
+  void selfDestruct() override {
+    Blaster::selfDestruct();
+    SaberBase::DoEffect(EFFECT_DESTRUCT, 0);
+    SetNextActionF(NEXT_ACTION_BLOW, hybrid_font.GetCurrentEffectLength());
+  }
+
+  // Clash to unjam or Enter/Exit Volume Menu.
+  void Clash(bool stab, float strength) override {
```

Comment:
same as override?

Reply:
Yeah, I guess in the base file's updating, they ended up the same now.
Removed from here. Thanks for the catch.